### PR TITLE
#1103 java.lang.NullPointerException

### PR DESCRIPTION
**Pull request by ramaiahkk**
java.lang.NullPointerException while decrypting due to value is null and still looking for cipher string. #1103

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@ramaiahkk Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=1104)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=1104) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@ramaiahkk Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=1104)!

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptor.java [hunk 1]
```diff
@@ -64,8 +64,11 @@ private Environment decrypt(Environment environment, TextEncryptorLocator encryp
 			for (Map.Entry<Object, Object> entry : new LinkedHashSet<>(map.entrySet())) {
 				Object key = entry.getKey();
 				String name = key.toString();
-				String value = entry.getValue().toString();
-				if (value.startsWith("{cipher}")) {
+				String value = null;
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptor.java]: This does not need to be outside the if, its not used anywhere else

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptor.java [hunk 2]
```diff
@@ -64,8 +64,11 @@ private Environment decrypt(Environment environment, TextEncryptorLocator encryp
...
+				if(entry.getValue() != null){
+				  value = entry.getValue().toString();		
+				}
+				if (value != null && value.startsWith("{cipher}")) {
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptor.java]: remove the `value` `null ` check and just keep this `if` wrapped in the above `if`

**Comment by ramaiahkk** [comment]
@ryanjbaxter Test added to check if null returns null

**Pull request merged**